File: src/main/java/eu/cessda/cvs/service/ConceptService.java
/*
 * Copyright © 2017-2021 CESSDA ERIC (support@cessda.eu)
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and limitations under the License.
 */

package eu.cessda.cvs.service;

import eu.cessda.cvs.service.dto.ConceptDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

/**
 * Service Interface for managing {@link eu.cessda.cvs.domain.Concept}.
 */
public interface ConceptService {

    /**
     * Save a concept.
     *
     * @param conceptDTO the entity to save.
     * @return the persisted entity.
     */
    ConceptDTO save(ConceptDTO conceptDTO);

    /**
     * Get all the concepts.
     *
     * @param pageable the pagination information.
     * @return the list of entities.
     */
    Page<ConceptDTO> findAll(Pageable pageable);


    /**
     * Get the "id" concept.
     *
     * @param id the id of the entity.
     * @return the entity.
     */
    Optional<ConceptDTO> findOne(Long id);

    /**
     * Delete the "id" concept.
     *
     * @param id the id of the entity.
     */
    void delete(Long id);

    /**
     * Get all the concepts given version
     *
     * @param versionId the version information.
     * @return the list of entities.
     */
    List<ConceptDTO> findByVersion(Long versionId);

    /**
     *  Get all the concepts for given notation
     * 
     * @param notation the concept notation
     * @return tjhe list of entities
     */
    List<ConceptDTO> getByNotation(String notation);

    /**
     * Search for the concept corresponding to the query.
     *
     * @param query the query of the search.
     *
     * @param pageable the pagination information.
     * @return the list of entities.
     */
    Page<ConceptDTO> search(String query, Pageable pageable);
}

File: src/main/webapp/app/entities/concept/concept.service.ts
/*
 * Copyright © 2017-2023 CESSDA ERIC (support@cessda.eu)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { HttpClient, HttpResponse } from '@angular/common/http';
import { Observable } from 'rxjs';

import { SERVER_API_URL } from 'app/app.constants';
import { createRequestOption, SearchWithPagination } from 'app/shared/util/request-util';
import { Concept } from 'app/shared/model/concept.model';

type EntityResponseType = HttpResponse<Concept>;
type EntityArrayResponseType = HttpResponse<Concept[]>;

@Injectable({ providedIn: 'root' })
export class ConceptService {
  public resourceUrl = SERVER_API_URL + 'api/concepts';
  public resourceSearchUrl = SERVER_API_URL + 'api/_search/concepts';

  constructor(protected http: HttpClient) {}

  create(concept: Concept): Observable<EntityResponseType> {
    return this.http.post<Concept>(this.resourceUrl, concept, { observe: 'response' });
  }

  update(concept: Concept): Observable<EntityResponseType> {
    return this.http.put<Concept>(this.resourceUrl, concept, { observe: 'response' });
  }

  find(id: number): Observable<EntityResponseType> {
    return this.http.get<Concept>(`${this.resourceUrl}/${id}`, { observe: 'response' });
  }

  query(req?: any): Observable<EntityArrayResponseType> {
    const options = createRequestOption(req);
    return this.http.get<Concept[]>(this.resourceUrl, { params: options, observe: 'response' });
  }

  delete(id: number): Observable<HttpResponse<object>> {
    return this.http.delete(`${this.resourceUrl}/${id}`, { observe: 'response' });
  }

  search(req: SearchWithPagination): Observable<EntityArrayResponseType> {
    const options = createRequestOption(req);
    return this.http.get<Concept[]>(this.resourceSearchUrl, { params: options, observe: 'response' });
  }
}
